feat: add non-disruptive esxi cert replacement (#121)
vmware/powershell-module-for-vmware-cloud-foundation-certificate-management: docs/documentation/functions/Install-VCFCertificate.md

Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -migratePowerOffVMs -vsanDataMigrationMode EnsureAccessibility -certificateDirectory F:\certificates -certificateFileExt ".cer"
```

This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path. When VMware Cloud Foundation 
version is earlier than 5.2, the ESXi host will enter maintenance mode with Migrate Power off VMs option enabled and vSAN data migration Mode set to `EnsureAccessibility`.

> ### EXAMPLE 3

Reviewer: ```suggestion
### Example 3
```

Author: addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba

```powershell
Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -certificateDirectory F:\certificates -certificateFileExt ".cer"
```

This example will install certificates for each ESXi host in cluster sfo-m01-cl01 in workload domain sfo-m01 from the provided path.  When VMware Cloud Foundation 
version is 5.2 or later, the vsanDataMigrationMode option no longer applied.  

> ### EXAMPLE 4

Reviewer: ```suggestion
### Example 4
```

Author: addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba

```powershell
Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -certificateDirectory F:\certificates -certificateFileExt ".cer" -uploadPrivateKey
```

This example will install private keys and certificates for each ESXi host in cluster sfo-m01-cl01 in workload domain sfo-m01 from the provided path.  The uploadprivatekey 

> parameter is only validate for VMware Cloud Foundation version is 5.2 or later.

Reviewer: ```suggestion
parameter is only available for VMware Cloud Foundation version is 5.2 or later.
```

Author: addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba

> ### EXAMPLE 5

Reviewer: ```suggestion
### Example 5
```

Author: addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba

```powershell
Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -migratePowerOffVMs -vsanDataMigrationMode EnsureAccessibility -certificateDirectory F:\certificates -certificateFileExt ".cer"
```

> This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path.  When VMware Cloud Foundation 

Reviewer: ```suggestion
This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path. When VMware Cloud Foundation 
```

Author: addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba

> version is earlier than 5.2, the ESXi host will enter maintenance mode with Migrate Power off VMs option enabled and vSAN data migration Mode set to EnsureAccessibility.

Reviewer: This reads a bit odd.

Author: pending will update this in a later.

Accept wildcard characters: False
```

### -uploadPrivateKey

> Option to upload a custom private key when performing the ESXi host certificate replacement (only valid for VCF 5.2 and later).

Reviewer: ```suggestion
Option to upload a custom private key when performing the ESXi host certificate replacement. Supported on VMware Cloud Foundation 5.2 or later.
```

Author: addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba
